Fix gnome notifications

zebreus opened the pull request:
This PR fixes notifications on gnome. Previously they were not shown, because `X-GNOME-UsesNotifications=true` was missing from the .desktop file. [Relevant documentation](https://teams.pages.gitlab.gnome.org/Websites/developer.gnome.org/documentation/tutorials/notifications.html)

SoongNoonien commented:
Hi!

> Previously they were not shown, ...

Where were they not shown?

SoongNoonien commented:
You mean here?
![Bildschirmfoto vom 2024-04-15 15-11-35](https://github.com/SoongNoonien/plattenalbum/assets/49582808/ef26f314-4871-4438-8a7e-f3ecd51f7dce)

zebreus commented:
I was building `plattenalbum` from source on NixOS and notifications were not showing up at all, even though I had them enabled in the plattenalbum settings. Adding `X-GNOME-UsesNotifications=true` seemed to fix the notifications. I tried it again today, and it works without `X-GNOME-UsesNotifications=true` so maybe it was just a fluke.

I am closing this PR for now.

SoongNoonien commented:
Did you receive a notification prior to the appearance in the settings? I'm not sure how exactly gnome determines what applications to show but xfce for example lists all apps that ever sent a notification. So maybe `X-GNOME-UsesNotifications=true` is needed to show an app even if it never sent a notification?

zebreus commented:
I have not looked into the notifications tab in the settings before I added `X-GNOME-UsesNotifications=true` because I did not know that that tab existed. After your comment I looked into there and plattenalbum had an entry.

I then rebuild plattenalbum without `X-GNOME-UsesNotifications=true` and deleted all related config from dconf (`/de/wagnermartin/Plattenalbum/` and `/org/gnome/desktop/notifications/application/de-wagnermartin-plattenalbum/`). I then started plattenalbum and the entry in the settings was recreated and notifications worked. However I did not restart my machine, so it may be that gnome still had the old desktop file cached in some way.

The [GNotification documentation](https://docs.gtk.org/gio/class.Notification.html#:~:text=The%20X%2DGNOME%2DUsesNotifications%20key,Control%20Center's%20'Notifications'%20panel) suggests that `X-GNOME-UsesNotifications=true` is needed or at least helpful for it showing up in the notification settings. But if it is required then I am confused why it apparently worked before. Maybe it creates the settings entry on the first actual notification, but that would not explain why it did not work for me initially...

I will try removing, rebooting, and then readding  `X-GNOME-UsesNotifications=true` later this week, maybe that gives us some clarity.

SoongNoonien commented:
> I have not looked into the notifications tab in the settings before I added X-GNOME-UsesNotifications=true because I did not know that that tab existed.

Ok, so maybe I didn't understand you correctly.

> I was building plattenalbum from source on NixOS and notifications were not showing up at all, even though I had them enabled in the plattenalbum settings.

So you actually meant you didn't get any notifications in the notification area of gnome? I only send a notification when the player window doesn't have the keyboard focus. Maybe this was the issue you observed?

zebreus commented:
> So you actually meant you didn't get any notifications in the notification area of gnome? I only send a notification when the player window doesn't have the keyboard focus. Maybe this was the issue you observed?

Yes, I did not get any notifications in the notification area or anywhere else. I tried it with and without keyboard focus, it did not change anything. However as stated above I am no longer able to reproduce that behaviour.
